CNV-52860: Passt binding for primary UDNs (#100075)
openshift/openshift-docs: virt/vm_networking/virt-connecting-vm-to-primary-udn.adoc

.Next steps
* xref:../../virt/vm_networking/virt-connecting-vm-to-primary-udn.adoc#virt-creating-udn-namespace-cli_virt-connecting-vm-to-primary-udn[Create namespaces that are associated with the cluster-scoped UDN]

[id="attaching-vm-to-primary-udn_{context}"]
== Attaching a virtual machine to the primary user-defined network

You can connect a virtual machine (VM) to the primary user-defined network (UDN) by requesting the pod network attachment, and configuring the interface binding.

{VirtProductName} supports the following network binding plugins to connect the network interface to the VM:

> `l2bridge`:: The Layer 2 bridge binding creates a direct Layer 2 connection between the VM's virtual network and the host's virtual switch, which is then attached to a physical interface or a VLAN. This binding places the VM on the same network as the host, allowing it to obtain an IP address from the same DHCP server and share the host's physical network connection.

Reviewer: This description is correct for secondary networks, but not for the primary UDN

```suggestion
`l2bridge`:: The Layer 2 bridge binding creates a direct Layer 2 connection between the VM's virtual interface and the virtual switch of the user-defined network.
```

`passt`:: The Plug a Simple Socket Transport (passt) binding provides a user-space networking solution integrates seamlessly with the pod network, providing better integration with the {product-title} networking ecosystem.
+
Passt binding has the following benefits:

* You can define readiness and liveness HTTP probes to configure VM health checks.

> * You can use Red Hat Advanced Cluster Security to monitor TCP traffic within the cluster.

Reviewer 2: With detailed insights.

> `passt`:: The Plug a Simple Socket Transport (passt) binding translates Layer 2 network frames from a VM into Layer 4 sockets on the host machine. It provides a user-space networking solution that avoids needing special privileges for VMs that need network access but should not be directly exposed to the host network's Layer 2. 

Reviewer: This is also inaccurate in the UDN context
```suggestion
`passt`:: The Plug a Simple Socket Transport (passt) binding provides a user-space networking solution that integrates seamlessly with the Pod's network, providing better integration with OpenShift networking ecosystem. 
```

> * You can define readiness and liveness probes to configure VM health checks.

Reviewer: ```suggestion
* You can define readiness and liveness HTTP probes to configure VM health checks.
```
